Samples/src/UnitTesting/CCNOTCircuits.qs

```diff
@@ -296,28 +225,29 @@ namespace Microsoft.Quantum.Samples.UnitTesting {
     operation CCNOT3 (control1 : Qubit, control2 : Qubit, target : Qubit) : Unit {
 
         body (...) {
-            using (ancillas = Qubit[1]) {
-                let ancilla = ancillas[0];
-                UpToPhaseCCNOT2(control1, control2, ancilla);
-                S(ancilla);
-                CNOT(ancilla, target);
-                H(ancillas[0]);
-                AssertProb([PauliZ], [ancilla], One, 0.5, "", 1E-10);
-
-                if (M(ancilla) == One) {
+            using (auxillaryQubit = Qubit()) {
+                UpToPhaseCCNOT2(control1, control2, auxillaryQubit);
+                S(auxillaryQubit);
+                CNOT(auxillaryQubit, target);
+                H(auxillaryQubit);
+                AssertProb([PauliZ], [auxillaryQubit], One, 0.5, "", 1E-10);
+
+                if (M(auxillaryQubit) == One) {
                     Controlled Z([control2], control1);
-                    X(ancilla);
+                    X(auxillaryQubit);
                 }
             }
         }
 
-        adjoint self;
-
         // fall back to a standard multiply controlled X Implementation
         controlled (controls, ...) {
             Controlled X(controls + [control1, control2], target);
         }
 
+        // NB: We cannot use "is Adj" or "adjoint auto" here, due to the use of
+        //     Microsoft.Quantum.Intrinsic.M, which does is not Adj. Instead, we
+        //     explicitly use the "self" generator.
+        adjoint self;
         controlled adjoint self;
```

Comment:
this line is not needed (no longer causes a warning) - either the adjoint self or controlled adjoint self declaration is sufficient